Web components 0.7.0

cairocoder01 opened the pull request:
This implements the web components that are ready in version 0.7.0 of the web components library. That includes the following components:

- `dt-text` (field type: `text`)
- `dt-textarea` (field type: `textarea`)
- `dt-date` (field type: `date`) (does not handle `datetime`)
- `dt-multi-text` (field type: `communication_channel`) (may be extended in the future to handle `link` as well)
- `dt-single-select` (field type: `key_select`)
- `dt-multi-select` (field type: `multi_select` when using typeahead display)
- `dt-multi-select-button-group` (field type: `multi_select` when using button display)
- `dt-tags` (field type: `tags`)
- `dt-connection` (field type: `connection`)
- `dt-button` - used within `dt-multi-select-button-group` but not replace within the theme at the moment

This has been tested on the contact detail and new contact screens. 

Bulk add of new posts has been removed upon discussion with @corsacca. Code is still there but link to it has been removed.

Javascript files of details and new screens have been cleaned to remove old code that is no longer necessary and has been tested after removal to ensure everything is still working as expected.

Future versions of the component library will implement more components and integrate them with the theme in future PRs.

cairocoder01 committed:
copy components code from next branch

cairocoder01 committed:
refactor render of components into separate class

cairocoder01 committed:
update components to v0.7.0 using upgraded Lit

cairocoder01 committed:
remove localized js files

localization is compiled into main index.js file

cairocoder01 committed:
implement dt-multi-select-button-group

cairocoder01 committed:
update baptism modal

cairocoder01 committed:
update dt-connection and finish baptism modal logic conversion

cairocoder01 committed:
clean up outdated details.js code

cairocoder01 committed:
update new post form with components

cairocoder01 committed:
Merge remote-tracking branch 'origin/develop' into wc-beta

cairocoder01 committed:
remove bulk add option

cairocoder01 committed:
components v0.7.0

cairocoder01 committed:
fix phpcs

corsacca commented:
Awesome work!
Here is an unordered list of comments. Some bugs, some styling consistency and preferences.

There is an empty <p> tag under the contact status, adding uneeded margin.
<img width="300" alt="image" src="https://github.com/user-attachments/assets/b4a597f6-d8b0-4af5-a7e8-b8a34a769cee" />

Multiselect buttons font size is bigger than before and bigger than surrounding elements
<img width="359" alt="image" src="https://github.com/user-attachments/assets/d805c3ab-316e-4110-bbc4-d124f0a8f670" />

Multiselect items without icons have extra whitespace:
<img width="410" alt="image" src="https://github.com/user-attachments/assets/b5e0340c-24f8-4c95-a146-58434e9deea0" />

Missing padding/margin under multiselect buttons before next field:
<img width="401" alt="image" src="https://github.com/user-attachments/assets/cac744b0-696b-48d6-8cb3-7e1d0e995fed" />

X button on first empty com channel seems confusing
<img width="405" alt="image" src="https://github.com/user-attachments/assets/66cb0be6-53e6-4bb8-8d03-ebeff13c54f3" />

Cannot unselect key select back to "empty" value:
<img width="411" alt="image" src="https://github.com/user-attachments/assets/6515e4dc-3665-4e2e-b93c-74e30c7a6474" />

After setting baptism date, a modal opens and i get this error:
Trying to access array offset on null in /wp-content/themes/disciple-tools-theme/dt-core/utilities/dt-components.php on line 53

Com field. Adding a value and then clicking remove clears clears the field, but on refresh it value is there again. After refresh the remove button works.
Some issue for the links field.
<img width="397" alt="image" src="https://github.com/user-attachments/assets/234a9810-8860-430d-b449-529d724e145a" />


I see this one is for 0.8. Nice. I feel the + button to add is missing from connection and tag fields:
https://github.com/DiscipleTools/disciple-tools-web-components/issues/105

cairocoder01 committed:
tidy up field spacing on detail screen

cairocoder01 committed:
add empty value to select fields when applicable

cairocoder01 committed:
modal logic for contact status reason

cairocoder01 committed:
re-bind multi-text in order to delete new values by key

cairocoder01 committed:
components v0.7.1

cairocoder01 commented:
@corsacca 
Thanks for the thorough review. All of your comments should be address with these commits and v0.7.1 of the component library.

- `dt-single-select` had extra bottom margin on the select element, which is fixed. And overall status had a `<p>` tag that has been changed to `<div>` to prevent the extra margin intended for paragraphs
- `dt-multi-select-button-group`
    - fix font and padding sizing
    - remove empty icon space
- fixed spacing between all fields. No margin on components themselves. Margin added to cells in tiles.
- fix error when opening baptism modal
- add empty values for single selects when appropriate
- multi-text: did a re-bind of value after saving so new values can be deleted by key
- remove 'x' from first comm channel

corsacca committed:
Fix post type not being available for a field

corsacca commented:
Thanks @cairocoder01!

corsacca commented:
Can we consider leaving in the listeners in the details.js?
Some plugins like the Coaching checklist depend on those:
<img width="406" alt="image" src="https://github.com/user-attachments/assets/9dcbe7f5-84e8-4180-8b3a-94bd8b72eb83" />

corsacca commented:
In merging records, fields should be readonly:
<img width="416" alt="image" src="https://github.com/user-attachments/assets/42998db8-6f94-483b-accc-9a9b3af9a150" />

now:
It is now obvious that these are disabled:
<img width="453" alt="image" src="https://github.com/user-attachments/assets/9148bd92-291d-4f1d-af9e-95696992d1c4" />

And com channel and multiselect fields are editable, connect fields have eternal spinner, etc.

I think this is something to do with the disabled state?

cairocoder01 committed:
update components on merge-post-details

cairocoder01 committed:
include legacy js on details page

cairocoder01 committed:
update field spacing on new record page

cairocoder01 committed:
use shared render function for overall_status

field ID prefix was not being utilized correctly for the merge records page

cairocoder01 committed:
update components in bulk update of list

cairocoder01 committed:
components v0.7.2

cairocoder01 committed:
Merge remote-tracking branch 'origin/develop' into wc-beta

cairocoder01 commented:
I've added back in some of the old field init scripts for the detail screen. Each of them logs a deprecation warning in the console if they are triggered.

I went through the whole merge records process and updated it to work properly with the components.

I also went through the bulk edit feature on the list page, so that should be tested and working as well.

corsacca committed:
Merge branch 'refs/heads/develop' into web-components-0-7-0

# Conflicts:
#	dt-assets/build/css/style.min.css

corsacca committed:
update styles

corsacca committed:
update tile spacing

corsacca commented:
CHanged:
<img width="436" alt="image" src="https://github.com/user-attachments/assets/81bcdf85-f6a8-4ee1-8ecb-6ab273dedf1f" />

to

<img width="443" alt="image" src="https://github.com/user-attachments/assets/f686ff5d-d799-4fd4-99f7-4b583147bb88" />

corsacca commented:
@cairocoder01 I looks like number fields are not updating

corsacca committed:
fix field gap spacing

cairocoder01 committed:
add back text-input js for number field

corsacca commented:
Thanks @cairocoder01!

Which field types are still left to do in future PRs?
location, number

cairocoder01 commented:
Remaining to be implemented:

- number
- boolean
- datetime
- user_select
- location / location_meta
- link
- church health metrics

corsacca committed:
only apply tile gap to details pages

corsacca commented:
3 2 1 Components merge!
Thank you @cairocoder01!
